bigbluebutton-html5/imports/ui/components/plugins-engine/types.ts

```diff
@@ -0,0 +1,41 @@
+import * as React from 'react';
+import * as PluginSdk from 'bigbluebutton-html-plugin-sdk';
+
+export interface PluginProvidedStateProps {
+    uuid: string
+}
+
+export interface PluginsEngineComponentProps {
+    containerRef: React.RefObject<HTMLDivElement>
+}
+
+export interface PluginLoaderComponentProps {
+    uuid: string
+    containerRef: React.RefObject<HTMLDivElement>
+    loadedPlugins: React.MutableRefObject<number>
+    setLastLoadedPlugin: React.Dispatch<React.SetStateAction<HTMLScriptElement | undefined>>
+    pluginConfigSettings: PluginConfigSetting
+}
+
+export interface PluginEngineProps {
+    onReady: Function
+}
+
+export interface PluginConfigSetting {
+    name: string
+    url: string
+}
+
+export interface UniquePluginConfigSetting {
+    name: string
+    url: string
+    uuid: string
+}
```

Comment:
This name `UniquePluginConfigSetting` can be read as if you are making a deduplication of the plugins, but it's the same interface as `PluginConfigSetting` added with the `uuid` attribute.
Said that, I propose two changes:

1 - Instead of repeating `name` and `url` definitions, make this type extend from `PluginConfigSetting`.

2 - Rename it to `EffectivePluginConfig`